xbmc/addons/addoninfo/AddonInfo.cpp

```diff
@@ -284,27 +284,24 @@ std::vector<AddonInstanceId> CAddonInfo::GetKnownInstanceIds() const
     return singletonInstance;
 
   std::vector<AddonInstanceId> ret;
-
-  if (!XFILE::CDirectory::Exists(StringUtils::Format("special://profile/addon_data/{}/", m_id)))
-  {
-    ret.emplace_back(ADDON_FIRST_INSTANCE_ID);
-    return ret;
-  }
-
   const std::string searchPath = StringUtils::Format("special://profile/addon_data/{}/", m_id);
-  CFileItemList items;
-  XFILE::CDirectory::GetDirectory(searchPath, items, ".xml", XFILE::DIR_FLAG_NO_FILE_DIRS);
 
-  for (const auto& item : items)
+  if (XFILE::CDirectory::Exists(searchPath))
   {
-    const std::string startName = "instance-settings-";
-    std::string filename = URIUtils::GetFileName(item->GetPath());
-    if (StringUtils::StartsWithNoCase(URIUtils::GetFileName(item->GetPath()), startName))
+    CFileItemList items;
+    XFILE::CDirectory::GetDirectory(searchPath, items, ".xml", XFILE::DIR_FLAG_NO_FILE_DIRS);
+
+    for (const auto& item : items)
     {
-      URIUtils::RemoveExtension(filename);
-      const std::string uid = filename.substr(startName.length());
-      if (!uid.empty() && StringUtils::IsInteger(uid))
-        ret.emplace_back(std::atoi(uid.c_str()));
+      const std::string startName = "instance-settings-";
+      std::string filename = URIUtils::GetFileName(item->GetPath());
+      if (StringUtils::StartsWithNoCase(URIUtils::GetFileName(item->GetPath()), startName))
+      {
+        URIUtils::RemoveExtension(filename);
+        const std::string uid = filename.substr(startName.length());
```

Comment:
Not possible to test currently, but maybe usable as `std::stringview` to prevent the copy about. See https://stackoverflow.com/questions/40616953/divide-a-stdstring-without-copying

```cpp
        URIUtils::RemoveExtension(filename);
        const std::stringview uid = std::string_view(filename.data() + startName.length());
        if (!uid.empty() && StringUtils::IsInteger(uid.data()))
          ret.emplace_back(std::atoi(uid.data()));
```